libs/polkadot-api-provider/src/webb-provider/chain-query.ts

```diff
@@ -19,59 +20,66 @@ export class PolkadotChainQuery extends ChainQuery<WebbPolkadot> {
     return block.block.header.number.toNumber();
   }
 
-  private async getTokenBalanceByAssetId(assetId: string): Promise<string> {
-    const activeAccount = await this.inner.accounts.activeOrDefault;
-    if (activeAccount) {
-      // If the assetId is not 0, query the orml tokens
-      if (assetId !== '0') {
-        let tokenAccountData: OrmlTokensAccountData;
-
-        try {
-          tokenAccountData = await this.inner.api.query.tokens.accounts(
-            activeAccount.address,
-            assetId
-          );
-        } catch (e) {
-          // It is possible that we have connected to a chain that is not setup with orml tokens.
-          return '';
+  private getTokenBalanceByAssetId(assetId: string): Observable<string> {
+    return this.inner.newBlock.pipe(
+      switchMap(async () => {
+        const activeAccount = await this.inner.accounts.activeOrDefault;
+        if (activeAccount) {
+          // If the assetId is not 0, query the orml tokens
+          if (assetId !== '0') {
+            let tokenAccountData: OrmlTokensAccountData;
+
+            try {
+              tokenAccountData = await this.inner.api.query.tokens.accounts(
+                activeAccount.address,
+                assetId
+              );
+            } catch (e) {
+              // It is possible that we have connected to a chain that is not setup with orml tokens.
+              return '';
```

Comment:
Shouldn't we handle the error properly here?

Reply:
we can open an issue for better holding of this,but don't think we should do it in this PR